pytorch_amazon_network_analysis/amazon_network_analysis.ipynb

```diff
@@ -0,0 +1,753 @@
+    "for result in results:\n",
+    "    print(result)\n",
+    "\n",
+    "for query in eval_queries:\n",
+    "    results = memgraph.execute_and_fetch(query.strip())\n",
+    "    for result in results:\n",
+    "        continue"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6acebec0-9c93-43f2-8de5-f0630455887f",
+   "metadata": {},
+   "source": [
+    "After our **stream** is done, we should probably do few more rounds of training and evaluation in order to have properly working model. We can do so with following query:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 37,
+   "id": "fa94a5f4-fdc1-4323-8744-8208eabbf412",
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    },
+    "vscode": {
+     "languageId": "python"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "num_of_epochs = 5\n",
+    "\n",
+    "results = memgraph.execute_and_fetch(\n",
+    "    f\"\"\"\n",
+    "    CALL tgn.train_and_eval({num_of_epochs}) YIELD *\n",
+    "    RETURN epoch_num, batch_num, precision, batch_process_time, batch_type ORDER BY epoch_num, batch_num;\n",
+    "    \"\"\"\n",
+    ")\n",
+    "\n",
+    "for result in results:\n",
+    "    continue\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f9e6a412-8eba-4131-a8d6-3041ce1482b7",
+   "metadata": {},
+   "source": [
+    "Now, let's get all results and afterwards we will do some **plotting** to check whether our **precision** increases epoch after epoch."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 40,
+   "id": "a90307e3-3737-4796-b5a4-ac861a136faf",
+   "metadata": {
+    "vscode": {
+     "languageId": "python"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "from collections import defaultdict\n",
+    "import matplotlib.pyplot as plt\n",
+    "import numpy as np\n",
+    "\n",
+    "results_train_dict = defaultdict(list)\n",
+    "results_eval_dict = defaultdict(list)\n",
+    "\n",
+    "results = memgraph.execute_and_fetch(\n",
+    "    \"\"\"\n",
+    "        CALL tgn.get_results() YIELD  epoch_num, batch_num, precision, batch_process_time, batch_type\n",
+    "        RETURN epoch_num, batch_num, precision, batch_process_time, batch_type ORDER BY epoch_num, batch_num;\n",
+    "    \"\"\"\n",
+    ")\n",
+    "\n",
+    "for result in results:\n",
+    "    if result['batch_type'] == 'Train':\n",
+    "        results_train_dict[result['epoch_num']].append(result['precision'])\n",
+    "    else:\n",
+    "        results_eval_dict[result['epoch_num']].append(result['precision'])\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f98a46d5-3c58-43db-b9cd-ac91c163b562",
+   "metadata": {},
+   "source": [
+    "Now when we have collected results, let's first plot average `accuracy` of `train` batches inside epoch, and average `accuracy` of `eval` batches inside epoch. We can do that since every batch is the same size. (**Remember**: `TGN` works with predefined batch size.)"
```

Comment:
```suggestion
    "Now when we have collected the results, let's first plot the average `accuracy` of `train` batches inside epoch, and the average `accuracy` of `eval` batches inside epoch. We can do that since every batch is the same size. (**NB**: `TGN` uses a predefined batch size.)"
```